lib/Prelude.fram

```diff
@@ -17,107 +21,14 @@ pub let snd (_, y) = y
-pub method fn (>=) = ge
-pub method fn (<)  = lt
-pub method fn (<=) = le
-
-pub method fn (&&&) = land
-pub method fn (^^^) = lxor
-pub method fn (|||) = lor
-pub method fn (<<)  = shiftl
-pub method fn (>>)  = shiftr
-pub method fn (>>>) = ashiftr
-
-pub method equal = (extern dbl_eqInt  : Int -> Int -> Bool) self
-pub method neq   = (extern dbl_neqInt : Int -> Int -> Bool) self
-pub method gt    = (extern dbl_gtInt  : Int -> Int -> Bool) self
-pub method lt    = (extern dbl_ltInt  : Int -> Int -> Bool) self
-pub method ge    = (extern dbl_geInt  : Int -> Int -> Bool) self
-pub method le    = (extern dbl_leInt  : Int -> Int -> Bool) self
-
-pub method toString = (extern dbl_intToString : Int -> String) self
-
-pub method add = (extern dbl_addInt : Int -> Int -> Int) self
-pub method sub = (extern dbl_subInt : Int -> Int -> Int) self
-pub method mul = (extern dbl_mulInt : Int -> Int -> Int) self
-
-pub method div {`re : {type X} -> Unit ->[|_] X} (n : Int) =
-  if n.equal 0 then `re ()
-  else (extern dbl_divInt : Int -> Int -> Int) self n
-
-pub method mod {`re : {type X} -> Unit ->[|_] X} (n : Int) =
-  if n.equal 0 then `re ()
-  else (extern dbl_modInt : Int -> Int -> Int) self n
-
-pub method land = (extern dbl_andInt : Int -> Int -> Int) self
-pub method lor  = (extern dbl_orInt  : Int -> Int -> Int) self
-pub method lxor = (extern dbl_xorInt : Int -> Int -> Int) self
-
-pub method shiftl  = (extern dbl_lslInt : Int -> Int -> Int) self
-pub method shiftr  = (extern dbl_lsrInt : Int -> Int -> Int) self
-pub method ashiftr = (extern dbl_asrInt : Int -> Int -> Int) self
-
-pub method add = (extern dbl_strCat : String -> String -> String) self
-
-pub method equal = (extern dbl_eqStr  : String -> String -> Bool) self
-pub method neq   = (extern dbl_neqStr : String -> String -> Bool) self
-pub method gt    = (extern dbl_gtStr  : String -> String -> Bool) self
-pub method lt    = (extern dbl_ltStr  : String -> String -> Bool) self
-pub method ge    = (extern dbl_geStr  : String -> String -> Bool) self
-pub method le    = (extern dbl_leStr  : String -> String -> Bool) self
-
-pub method length = (extern dbl_strLen : String -> Int) self
-pub method get {`re : {type X} -> Unit ->[|_] X, self : String} (n : Int) =
-  if n >= 0 && n < self.length then
-    (extern dbl_strGet : String -> Int -> Char) self n
-  else `re ()
-
-pub method makeString {`re : {type X} -> Unit ->[|_] X, self : String}
-    (n : Int) =
-  if n >= 0 && n < 256 then
-    (extern dbl_strMake : Int -> String) n
-  else `re ()
-
-pub method toList {self : String} =
-  let getChar = extern dbl_strGet : String -> Int -> Char in
-  let rec iter (n : Int) acc = 
-    if n == 0 then 
-      acc
-    else 
-      iter (n - 1) (getChar self (n - 1) :: acc) 
-  in iter self.length []
-
 pub let charListToStr = (extern dbl_chrListToStr : List Char -> String)
 
-pub method code {self : Char} = 
-  extern dbl_chrCode : Char -> Int
-
 pub let chr {`re : {type X} -> Unit ->[|_] X} (n : Int) = 
```

Comment:
I agree. However, I'm against moving it into `/Base/Char/` as the main purpose of this module is to contain **methods** that might be used without importing `Char` explicitly. We should create a separate module `Char` for that. I thing it can be done as a separate task.